sdk/storage/storage-file-share/test/fileserviceclient.spec.ts

```diff
@@ -380,4 +385,83 @@ describe("FileServiceClient", () => {
     assert.ok(typeof result.requestId);
     assert.ok(result.requestId!.length > 0);
   });
+
+  it("ListShares with deleted share", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    await shareClient.create();
+    await shareClient.delete();
+
+    let found = false;
+    for await (const share of serviceClient.listShares({ includeDeleted: true })) {
+      if (share.name == shareClient.name) {
+        found = true;
+        assert.ok(share.version);
+        assert.ok(share.deleted);
+      }
+    }
+    assert.ok(found);
+  });
+
+  it("Undelete share positive", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    await shareClient.create();
+    await shareClient.delete();
+
+    let found = false;
+    let shareDeleted: ShareItem | undefined;
+    for await (const share of serviceClient.listShares({ includeDeleted: true })) {
+      if (share.name == shareClient.name) {
+        found = true;
+        assert.ok(share.version);
+        assert.ok(share.deleted);
+        assert.ok(share.properties.deletedTime);
+        assert.ok(share.properties.remainingRetentionDays);
+
+        shareDeleted = share;
+      }
+    }
+    assert.ok(found);
+    assert.ok(shareDeleted);
+
+    // Await share to be deleted.
+    await delay(30 * 1000);
+
+    const restoredShareClient = await serviceClient.undeleteShare(
+      shareDeleted!.name,
+      shareDeleted!.version!
+    );
+    await restoredShareClient.getProperties();
+
+    await restoredShareClient.delete();
+  });
+
+  it("Undelete share negative", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    const invalidVersion = "01D60F8BB59A4652";
+
+    try {
+      await serviceClient.undeleteShare(shareClient.name, invalidVersion);
+      assert.fail("Expecting an error in undelete share with invalid version.");
+    } catch (error) {
+      assert.ok((error.statusCode as number) === 404);
```

Comment:
Shouldn't the `error.statusCode` be a number already?

Reply:
:) error here is without strong type, so as number is necessary.

---
In reply to: [475543258](https://github.com/Azure/azure-sdk-for-js/pull/10792#discussion_r475543258) [](ancestors = 475543258)